src/main/java/com/flightstats/hub/webhook/WebhookLeader.java

```diff
@@ -262,12 +259,8 @@ private void send(ContentPath contentPath) throws InterruptedException {
                 long start = System.currentTimeMillis();
                 boolean shouldGoToNextItem = retryer.send(webhook, contentPath, webhookStrategy.createResponse(contentPath));
                 statsdReporter.time("webhook", start, "name:" + webhook.getName());
-                if (shouldGoToNextItem) {
-                    if (increaseLastUpdated(contentPath)) {
-                        if (!deleteOnExit.get()) {
-                            clusterCacheDao.setIfNewer(contentPath, webhook.getName(), WEBHOOK_LAST_COMPLETED);
-                        }
-                    }
+                if (shouldGoToNextItem && increaseLastUpdated(contentPath)) {
+                    clusterCacheDao.setIfNewer(contentPath, webhook.getName(), WEBHOOK_LAST_COMPLETED);
                 }
                 keysInFlight.remove(webhook.getName(), contentPath);
```

Comment:
Just a consideration, I am probably not grokking all going on here but it looks to me logically like this should move inside of the if block?

Reply:
...I have no idea. I'll see if I can make any sense of it. May leave it for now so as to not change behavior.

Reply:
I looked into this a little more and keysInFlight tracks the stuff that's being sent *right now*. It looks like it has at least one purpose: making sure that every item that's being delivered finishes prior to turning off the webhook.
So having it outside the if block might make sense? I think. There are no unit-type tests around this, I think, so I'm going to leave it alone.